src/lib/builders/menubar/create.ts

```diff
@@ -71,15 +75,13 @@ export function createMenubar(props?: CreateMenubarProps) {
 		},
 		action: (node: HTMLElement) => {
 			const menuTriggers = Array.from(node.querySelectorAll('[data-melt-menubar-trigger]'));
-			if (!menuTriggers.length || !isHTMLElement(menuTriggers[0])) return {};
+			if (!isHTMLElement(menuTriggers[0])) return {};
 			menuTriggers[0].tabIndex = 0;
 
-			const menus = Array.from(node.querySelectorAll('[data-melt-menubar-menu]')).filter(
+			scopedMenus = Array.from(node.querySelectorAll('[data-melt-menubar-menu]')).filter(
```

Comment:
This only runs once, meaning if any menu is added or destroyed later on, scopedMenus value won't be accurate.

Either we constantly keep it updated by adding handlers on the trigger's action, or we instead create a function `getScopedMenus`, that works by getting the root menubarr via its id.